src/svc_rqst.c

```diff
@@ -643,6 +643,11 @@ svc_rqst_rearm_events_locked(SVCXPRT *xprt, uint16_t ev_flags)
 	tracepoint(xprt, rearm, __func__, __LINE__, xprt, ev_flags);
 #endif /* USE_LTTNG_NTIRPC */
 
+	const bool is_xprt_destroyed = xprt->xp_flags & (ev_flags | SVC_XPRT_FLAG_DESTROYED);
+	/* MUST follow the destroyed check above */
+	const bool is_rec_shutdown = !is_xprt_destroyed && (
+		sr_rec->ev_flags & SVC_RQST_FLAG_SHUTDOWN);
+
```

Comment:
How is this any safer? You examine xp_flags and ev_flags at the top of a code block that does nothing that would change the viability of xprt or sr_rec before you test at the bottom of the code block.

Reply:
The problem is inside the `__warnx`, where we access `sr_rec` regardless of the XPRT state. In this fix we guarantee it won't be accessed if `is_xprt_destroyed`.

Reply:
@ffilz friendly ping :)